Utilities/dronekit_wrappers.py

```diff
@@ -0,0 +1,93 @@
+from sys import stdout
+from time import sleep, time
+
+# Ours
+import constants as c
+
+def get_velocity_message(message_factory, (velocity_x, velocity_y, velocity_z)):
+    """
+    Returns a DroneKit object that represents a mavlink message which moves
+    the drone at a certain velocity.
+
+    Parameters
+    ----------
+    message_factory: dronekit.vehicle.message_factory
+        Used to create the message
+    (velocity_x, velocity_y, velocity_z): (Double, Double ,Double)
+        The vector to travel along
+
+    Returns:
+    ----------
+    MAVLink_message (a DroneKit object)
+    """
+    return message_factory.set_position_target_global_int_encode(
+            0,       # time_boot_ms (not used)
+            0, 0,    # target system, target component
+            mavutil.mavlink.MAV_FRAME_GLOBAL_RELATIVE_ALT_INT, # frame
+            0b0000111111000111, # type_mask (only speeds enabled)
+            0, # lat_int - X Position in WGS84 frame in 1e7 * meters
+            0, # lon_int - Y Position in WGS84 frame in 1e7 * meters
+            0, # alt - Altitude in meters in AMSL altitude(not WGS84 if absolute or relative)
+            # altitude above terrain if GLOBAL_TERRAIN_ALT_INT
+            velocity_x, # X velocity in NED frame in m/s
+            velocity_y, # Y velocity in NED frame in m/s
+            velocity_z, # Z velocity in NED frame in m/s
+            0, 0, 0, # afx, afy, afz acceleration (not supported yet, ignored in GCS_Mavlink)
+            0, 0)    # yaw, yaw_rate (not supported yet, ignored in GCS_Mavlink)
+   
+
+def set_attitude(vehicle, roll_angle = 0.0, pitch_angle = 0.0, yaw_rate = 0.0, thrust = 0.5):
+    """
+    NOTE: CURRENTLY NOT USED
+    The follow comments are from the DroneKit website:
+
+    Note that from AC3.3 the message should be re-sent every second (after about 3 seconds
+    with no message the velocity will drop back to zero). In AC3.2.1 and earlier the specified
+    velocity persists until it is canceled. The code below should work on either version
+    (sending the message multiple times does not cause problems).
+
+    The roll and pitch rate cannot be controllbed with rate in radian in AC3.4.4 or earlier,
+    so you must use quaternion to control the pitch and roll for those vehicles.
+    """
+    # Thrust >  0.5: Ascend
+    # Thrust == 0.5: Hold the altitude
+    # Thrust <  0.5: Descend
+    msg = vehicle.message_factory.set_attitude_target_encode(
+        0, # time_boot_ms
+        1, # Target system
+        1, # Target component
+        0b00000000, # Type mask: bit 1 is LSB
+        to_quaternion(roll_angle, pitch_angle), # Quaternion
+        0, # Body roll rate in radian
+        0, # Body pitch rate in radian
+        radians(yaw_rate), # Body yaw rate in radian
+        thrust  # Thrust
+    )
+    vehicle.send_mavlink(msg)
+
+def to_quaternion(roll = 0.0, pitch = 0.0, yaw = 0.0):
+    """
+    NOTE: CURRENTLY NOT USED
+
+    Convert degrees to quaternions
```

Comment:
Cite a source for the formulas applied for quaternions here.

Reply:
This is another one that was around before I joined flight. I was able to find an explanation on wikipedia at https://en.wikipedia.org/wiki/Conversion_between_quaternions_and_Euler_angles#Source_Code and will put this link in the code.

Reply:
This was one of the functions that was around before flight. I was able to find an explanation on wikipedia: https://en.wikipedia.org/wiki/Conversion_between_quaternions_and_Euler_angles#Source_Code